designs/package-spec.md

```diff
@@ -0,0 +1,78 @@
+# Package Controller Configuration
+
+## Introduction
+
+**Problem:** currently the ability to configure the package controller is very limited and requires code changes
+
+Code changes currently have to be made to configure the package controller during cluster creation or using the install command. Changes need to be coordinated between the CLI and the package controller helm chart to alter configuration. This makes it more difficult to support things like various private registry configurations.
+
+### Tenets
+
+* ***Flexible:*** reduce dependency of the CLI on helm chart.
+
+### Goals and Objectives
+
+As an EKS-A cluster administrator I want to configure the package controller for:
+
+* Harbor using proxy cache
+* Harbor using replication
+* Harbor as a private registry using the import images command
+* Configure ECR token refresher
+* Configure for kubelet private registry authentication
+* Support proxy configuration
+* Support package controller installation from alternate registries
+* Disable installation of package controller
+* Allow for future configuration options
+
+### Statement of Scope
+
+**In scope**
+* All of the above
+
+## Current state
+The package controller can be used currently with:
+1. Proxy configurations
+1. ECR token refresher enabled/disabled
+1. Installation of the package controller from alternate registries
+
+## Overview of Solution
+Add a section to the cluster specification to enable/disable the package controller installation and pass configuration values.
+
+## Solution Details
+
+Add a packageController section to the cluster specification for example to use development/staging builds:
+```yaml
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: Cluster
+metadata:
+  name: billy
+spec:
+  packageController:
+    enable: true
+    values: |-
+      defaultRegistry: public.ecr.aws/l0g8r8j6
+      defaultImageRegistry: 857151390494.dkr.ecr.us-west-2.amazonaws.com
+```
+
+These values will be passed to the helm chart during installation just like `--set defaultRegistry=public.ecr.aws/l0g8r8j6` for example.
+
+## Implementation
+
+The proposed design:
+```
+// PackageController defines configuration of the package controller.
+type PackageController struct {
+    // +kubebuilder:validation:Optional
+    // +kubebuilder:default:="true"
+    // Enable installation of the package controller.
+    Enable bool `json:"enable,omitempty"`
```

Comment:
We can leave this discussion for the implementation PR, but since you added the go struct here: this is going to be difficult to default to true. After parsing the yaml file, if not set, this field will be false. But we won't know if it wasn't set (hence, should be defaulted to to true) or it was explicitly set to false.

One alternative would be to have it as `*bool`, although that makes accessing it in the code a bit annoying. Given this is probably going to be read only once in the codepath, it's probably an ok tradeoff.

Reply:
Done.